java/src/test/java/com/adhoc/flight/TestAdhocFlightClient.java

```diff
@@ -38,141 +40,141 @@
-        // Create FlightClient connecting to Drmeio.
-        createBasicFlightClient(HOST, PORT, USERNAME, PASSWORD);
-
-        // Create table
-        client.runQuery(CREATE_TABLE, true);
-
-        // Select
-        final CallHeaders callHeaders = new FlightCallHeaders();
-        callHeaders.insert("schema", DEFAULT_SCHEMA_PATH);
-        final HeaderCallOption callOption = new HeaderCallOption(callHeaders);
-        client.runQuery(SIMPLE_QUERY_NO_SCHEMA, callOption, true);
-
-        // Drop table
-        client.runQuery(DROP_TABLE, true);
-    }
-
-    @Test
-    public void testBadHostname() {
-        final FlightRuntimeException fre = assertThrows(FlightRuntimeException.class,
-                () -> createBasicFlightClient("1.1.1.1", PORT, USERNAME, PASSWORD));
-        assertEquals(FlightStatusCode.UNAVAILABLE, fre.status().code());
-    }
-
-    @Test
-    public void testBadPort() {
-        final FlightRuntimeException fre = assertThrows(FlightRuntimeException.class,
-                () -> createBasicFlightClient(HOST, 1111, USERNAME, PASSWORD));
-        assertEquals(FlightStatusCode.UNAVAILABLE, fre.status().code());
-    }
-
-    @Test
-    public void testBadPassword() {
-        final FlightRuntimeException fre = assertThrows(FlightRuntimeException.class,
-                () -> createBasicFlightClient(HOST, PORT, USERNAME, "BAD_PASSWORD"));
-        assertEquals(FlightStatusCode.UNAUTHENTICATED, fre.status().code());
-    }
-
-    @Test
-    public void testNonExistentUser() {
-        final FlightRuntimeException fre = assertThrows(FlightRuntimeException.class,
-                () -> createBasicFlightClient(HOST, PORT, "BAD_USER", PASSWORD));
-        assertEquals(FlightStatusCode.UNAUTHENTICATED, fre.status().code());
-
-    }
+  private static final String HOST = "localhost";
+  private static final int PORT = 32010;
+  private static final String USERNAME = "dremio";
+  private static final String PASSWORD = "dremio123";
+  public static final String SIMPLE_QUERY = "select * from (VALUES(1,2,3),(4,5,6))";
+
+  public static final String DEFAULT_SCHEMA_PATH = "$scratch";
+  public static final String DEFAULT_ROUTING_TAG = "test-routing-tag";
+  public static final String DEFAULT_ROUTING_QUEUE = "Low Cost User Queries";
+
+  public static final String CREATE_TABLE = "create table $scratch.simple_table as " + SIMPLE_QUERY;
+  public static final String CREATE_TABLE_NO_SCHEMA = "create table $scratch.simple_table as " + SIMPLE_QUERY;
+  public static final String SIMPLE_QUERY_NO_SCHEMA = "SELECT * FROM simple_table";
+  public static final String DROP_TABLE = "drop table $scratch.simple_table";
+
+  private AdhocFlightClient client;
+  private BufferAllocator allocator;
+
+  @Before
+  public void setup() {
+    allocator = new RootAllocator(Long.MAX_VALUE);
+  }
+
+  @After
+  public void shutdown() throws Exception {
+    allocator.getChildAllocators().forEach(BufferAllocator::close);
```

Comment:
I think closing the root allocator closes child allocators and this will result in an error.

Reply:
This was the only way of closing the allocator. When I tried to close it with Autoclosabels.close()  it gave erros that children were open.

Reply:
This seems wrong. If the FlightClient created any child allocators it should really free them up on close. Seems like a bug in the FlightClient itself then.

Previously, were you closing the FlightClient then closing the allocator?

Reply:
It tried both way, closing the client and then the allocator and vice-versa